source/h2_frames.c

```diff
@@ -144,11 +144,11 @@ const char *aws_http2_error_code_to_str(enum aws_http2_error_code h2_error_code)
 }
 
 struct aws_h2err aws_h2err_from_h2_code(enum aws_http2_error_code h2_error_code) {
-    AWS_PRECONDITION(h2_error_code > AWS_HTTP2_ERR_NO_ERROR && h2_error_code < AWS_HTTP2_ERR_COUNT);
+    AWS_PRECONDITION(h2_error_code >= AWS_HTTP2_ERR_NO_ERROR && h2_error_code < AWS_HTTP2_ERR_COUNT);
 
     return (struct aws_h2err){
         .h2_code = h2_error_code,
-        .aws_code = AWS_ERROR_HTTP_PROTOCOL_ERROR,
+        .aws_code = h2_error_code == AWS_HTTP2_ERR_NO_ERROR ? AWS_ERROR_SUCCESS : AWS_ERROR_HTTP_PROTOCOL_ERROR,
     };
 }
```

Comment:
maybe don't modify this function
When the user sends RST_STREAM just build the aws_h2err manually:
```
struct aws_h2err h2err = {
    .h2_code = http2_error_from_user,
    .aws_code = AWS_ERROR_HTTP_RST_STREAM_SENT,
};
```